src/DI/MonologExtension.php

```diff
@@ -169,9 +175,13 @@ public function beforeCompile()
 			$logger->addSetup('pushProcessor', ['@' . $serviceName]);
 		}
 
-		$config = $this->getConfig(['registerFallback' => empty($handlers)] + $this->getConfig($this->defaults));
+		// This part of code used deprecated methods, that were removed in Nette 3.0
+		// Rewritten so that the output is same as before
```

Comment:
Do we need to have completely the same behavior? This is going to be new major version anyway so maybe reworking it to whatever it now the recommended way in Nette would be better.

Reply:
I think I can make the behavior unchanged, while using new recommended way in Nette.
I actually did it. The problem is, the new way does not pass the PHPStan check.
First I thought it's me making some error. But Nette 3.0 packages with new recommended way do not pass either: https://travis-ci.org/nette/application/jobs/524606313#L655
(having same type of errors on similar code)

What do you suggest?

Reply:
Where can I find the code?

Reply:
Done. Commit https://github.com/haltuf/Monolog/commit/d4e1348cded921f07eea8cec6c7fff919f92add0 pushed to @JanGalek's repository.
I managed to solve the PHPStan errors in the end by adding ->castTo('array') on the end of Schema definition.